saml/middlewares/load-user.js

```diff
@@ -0,0 +1,30 @@
+import   _         from 'lodash'             ;
+import   request   from 'superagent'         ;
+import { apiUrl  } from '#~/saml//config.js' ;
+
+export default  function () {
+    return async (req, res, next) => {
+        
+        try {
+            const { authenticationToken } = req.cookies;
+
+            if(!authenticationToken) return next();
+
+            req.user = await getAuthUser(authenticationToken)
+            next();
+        }
+        catch(err) {
+            delete req.user; // make sure anonymous on error
+            next();
```

Comment:
if the request has token, it should return 401 else go to next()

Reply:
as discussed marking resolved